### ISSUE TITLE
Running "amplify pull --yes" appears to cause upload/push.

### ISSUE THREAD

**Issue by BennettSmith** [aws-amplify/amplify-cli#5278]
**Describe the bug**
Our build script performs an `amplify pull --providers {filename} --frontend {filename} --amplify {filename} --yes`.  Recently we noticed that this is causing the environment to be pulled from the backend *and then uploaded/pushed again*.  If we remove the `--yes` argument, then we are prompted with the question about making changes. We answer `Y` to this, and the pull happens as we expect (i.e. it is not followed up with an upload/push).

**Amplify CLI Version**
We are running version `4.29.0` of the cli.

**To Reproduce**
Perform a pull as mentioned int he description. Try it with and without the `--yes` switch to see the difference in behavior.

**Expected behavior**
We expect that performing a pull will not cause any files to be uploaded or deployed to the environment on the backend.

**Desktop (please complete the following information):**
Running on a Mac with 10.15.6.
Node.js version is v10.21.0

**Comment by BennettSmith** [comment]
More information:

When I run the `amplify pull` command with `--yes` and it does the push, it doesn't even seem to be complete. One part of our app is to use the `PostConfirmation` trigger for Cognito. The GROUP environment variable ends up being blank after this unexpected push, as seen in the screen shot attached.

<img width="1039" alt="Screen Shot 2020-09-08 at 10 28 22 AM" src="https://user-images.githubusercontent.com/123463/92508979-59bf2200-f1be-11ea-9e98-70dd200080e8.png">

Being blank causes another failure when trying to use the backend environment. The error is valid, but should not happen if a push is done properly.

```
caught error: "userLambdaValidation(message: "PostConfirmation failed with error 2 validation errors detected: Value \'\' at \'groupName\' failed to satisfy constraint: Member must satisfy regular expression pattern: [\\p{L}\\p{M}\\p{S}\\p{N}\\p{P}]+; Value \'\' at \'groupName\' failed to satisfy constraint: Member must have length greater than or equal to 1.")"
```

**Comment by mrgrue** [comment]
+1

**Comment by blydewright** [comment]
+1. We have a CD process setup through Vercel and require the `--yes` headless option, but since it runs a push, it slows down the deploy time significantly as we have a large app with many resources.

**Comment by edwardfoyle** [comment]
This has been fixed in version 4.30.0 of the CLI. If you still want to do a push after a pull you can use the --forcePush option, but just --yes will no longer do a push.
